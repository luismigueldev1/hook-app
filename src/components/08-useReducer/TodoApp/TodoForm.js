import React from "react";
import PropTypes from "prop-types";
import { useForm } from "../../../hooks/useForm";

function TodoForm({ handleAddTodo }) {
  const [values, handleInputChange, reset] = useForm({
    task: "",
  });

  const handleSubmit = (event) => {
    event.preventDefault();

    if (values.task.trim().length <= 1) {
      return;
    }

    const newTodo = {
      id: new Date().getTime(),
      task: values.task,
      done: false,
    };
    handleAddTodo(newTodo);
    reset();
  };

  return (
    <>
      <h4>Agregar TODO</h4>
      <hr />

      <form onSubmit={handleSubmit}>
        <input
          className="form-control"
          type="text"
          name="task"
          placeholder="New task"
          autoComplete="off"
          value={values.task}
          onChange={handleInputChange}
        />
        <button
          type="submit"
          className="btn btn-outline-primary  btn-block mt-2"
        >
          Add New Task
        </button>
      </form>
    </>
  );
}

TodoForm.propTypes = {
  handleAddTodo: PropTypes.func.isRequired,
};

export default TodoForm;
