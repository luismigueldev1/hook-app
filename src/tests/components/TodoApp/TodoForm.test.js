import { shallow } from "enzyme";
import TodoForm from "../../../components/08-useReducer/TodoApp/TodoForm";

describe("Pruebas en <TodoForm/>", () => {
  const handleAddTodo = jest.fn();

  const wrapper = shallow(<TodoForm handleAddTodo={handleAddTodo} />);
  test("debe de renderizarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("No debe de llamar handleAddTodo", () => {
    const formSubmit = wrapper.find("form").prop("onSubmit");
    formSubmit({ preventDefault() {} });

    expect(handleAddTodo).toHaveBeenCalledTimes(0);
  });

  test("Debe de llamar la funcion handleAddTodo", () => {
    const value = "Aprender React";
    wrapper.find("input").simulate("change", {
      target: {
        name: "task",
        value: value,
      },
    });
    const formSubmit = wrapper.find("form").prop("onSubmit");
    formSubmit({ preventDefault() {} });

    expect(handleAddTodo).toHaveBeenCalledTimes(1);
    expect(handleAddTodo).toHaveBeenCalledWith(expect.any(Object));
    expect(handleAddTodo).toHaveBeenCalledWith({
      id: expect.any(Number),
      task: value,
      done: false,
    });

    expect(wrapper.find("input").prop("value")).toBe("");
  });
});
