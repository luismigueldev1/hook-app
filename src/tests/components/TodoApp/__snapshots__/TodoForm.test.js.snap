// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TodoForm/> debe de renderizarse correctamente 1`] = `
<Fragment>
  <h4>
    Agregar TODO
  </h4>
  <hr />
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="form-control"
      name="task"
      onChange={[Function]}
      placeholder="New task"
      type="text"
      value=""
    />
    <button
      className="btn btn-outline-primary  btn-block mt-2"
      type="submit"
    >
      Add New Task
    </button>
  </form>
</Fragment>
`;
